### PR TITLE
[InputStream] Add support to audio codec profiles /  ClearKey crypto type

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

In reference to opened PR #23524 this allow InputStream binary addons to add the codec profile to audio streams
mainly to display (by using PR #23524) the appropriate codec name in the audio track list

I have already predisposed interface with the new codec profiles that will be available with the next ffmpeg version
    DTSCodecProfileHDMAX
    DTSCodecProfileHDMAIMAX
    DDPlusCodecProfileAtmos
so when we will switch to it there is no need to rebuild all the addons but only set the appropriate profile conversion on kodi core

---

This PR include commits of #23540 to add ClearKey as a valid type for encryption scheme like existing Widevine, PlayReady and WisePlay.

## How has this been tested?

Builded locally just to test that the value is set, ISAdaptive implementation will be done after merged
## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
This conflicts with https://github.com/xbmc/xbmc/pull/23540 in that the version bumps conflict.

Can the changes be done in a single PR with 3 commits, the last being the version bump?

**Comment by phunkyfish** [comment]
The change itself looks to be ok. But it is customary to have at least one addon use the API change lined up with a PR before merging the Kodi.

**Comment by CastagnaIT** [comment]
I didn't prepare it yet because there are too many PRs opened on ISAdaptive that touch different parts of it, and no one that can review PR's, PR stand opened for more than a week (yesterday by luck glenn guy review it, but i had not time)
also the other "clearkey" PR its not implemented on ISAdaptive

fwir features cannot be merged on Beta so we need these on this last kodi Alpha

in next days or next week i will do the ISA PR for audio codecs profiles,
for the "clearkey" implementation on ISA it is possible that glennguy wants to wait after the big decrypter refactor PR be merged
It will take some time because it is a huge PR

> Can the changes be done in a single PR with 3 commits, the last being the version bump?

i can merge the "clearkey" PR with this one soon

**Comment by phunkyfish** [comment]
Ok, I understand. Let’s get the 2 PRs into one so there is a single verison change and we can go from there.

As this is solely for inputstreams I think we can forgo normal protocols and make an exception.

**Review by phunkyfish — APPROVED** [review]
Please clean up the clang format suggestions. Otherwise this is good.

**Comment by CastagnaIT** [comment]
just to let you know that i opened the PR on ISAdaptive to implement audio profiles and tested
clang has been applied

**Comment by phunkyfish** [comment]
Ok, I think we should go ahead and merge this. We can fix the Apple errors on ffmpegdirect afterwards.

**Comment by CastagnaIT** [comment]
ouch i forgot a change in the CDVDDemuxClient for profiles i need some time

**Comment by CastagnaIT** [comment]
done

File: xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/stream_codec.h [hunk 1]
```diff
@@ -141,6 +141,28 @@ extern "C"
     /// [chroma subsampling](https://en.wikipedia.org/wiki/Chroma_subsampling): 4:2:0, 4:2:2, 4:4:4
     /// see [AV1 specification](https://aomedia.org/av1/specification/)
     AV1CodecProfileProfessional,
+
+    // Audio codec profiles
+
+    AACCodecProfileMAIN,
```
Comment (AlwinEsch) [inline xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/stream_codec.h:147]: Only a bit about API descriptions missing on new added values :smirk:

File: xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/stream_codec.h [hunk 2]
```diff
@@ -141,6 +141,28 @@ extern "C"
...
+    AACCodecProfileLTP,
+    AACCodecProfileHE,
+    AACCodecProfileHEV2,
+    AACCodecProfileLD,
+    AACCodecProfileELD,
+    MPEG2AACCodecProfileLOW,
+    MPEG2AACCodecProfileHE,
+    DTSCodecProfile,
+    DTSCodecProfileES,
+    DTSCodecProfile9624,
```
Comment (AlwinEsch) [inline xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/stream_codec.h:159]: Specially here, as firstly I become a big thinking in mind whats "9624" means :sweat_smile::thinking: and comes then the "96 kHz / 24 Bit" in mind.

**Review by AlwinEsch — APPROVED** [review]
Mainly OK, only about docs, but can also leaved as is.

File: xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/stream_codec.h (hunk 1 above)
Comment (CastagnaIT) [reply xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon-instance/inputstream/stream_codec.h:148]: description added for each one

**Pull request merged**